crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -336,12 +348,15 @@ impl ServerBuilder {
                 config.connection.prom_url, config.connection.prom_port
             ))
         })?;
+
         let registry_service = mysten_metrics::start_prometheus_server(prom_addr);
         info!("Starting Prometheus HTTP endpoint at {}", prom_addr);
         let registry = registry_service.default_registry();
         registry
             .register(mysten_metrics::uptime_metric(
-                "graphql", version.0, "unknown",
+                "graphql",
+                Box::leak(Box::new(version.clone().to_string())),
```

Comment:
the `uptime_metric` requires the version argument to be of type `&'static str`, so I don't understand how to make this work without `Box::leak`
@amnn any tips here?

Reply:
That is quite unfortunate... Best solution I can think of is to make the Version you use a `static` field itself, so that you can get a `&'static Version` field to it, and have it store a computed full version field in it:

```rust
struct Version {
    year: u64,
    month: u8,
    patch: u64,
    sha: String,
    // Compute this in the constructor.
    full_version: String,
}

// In main.rs
static version: Version = Version::new(/* ... */)
```

Reply:
Or actually, seeing your changes to the `Version` type, having some redundancy there is also fine?

```
struct Version {
    year: &'static str,
    month: &'static str,
    patch: &'static str,
    sha: &'static str,
    full: &'static str,
```

Reply:
Thanks, this helped. I ended up using the `full` field!